Fig4_PyGMT_pandas.py

```diff
@@ -0,0 +1,63 @@
+import io
+
+import pandas as pd
+import pygmt
+import requests
+from pygmt.params import Box
+
+params = {
+    "format": "csv",
+    "starttime": "2000-01-01",
+    "endtime": "2025-10-30",
+    "mindepth": 70,
+    "minmagnitude": 5,
+    "orderby": "magnitude",
+}
+r = requests.get("https://earthquake.usgs.gov/fdsnws/event/1/query", params=params)
+df_eqs = pd.read_csv(io.StringIO(r.text))
+df_eqs = df_eqs[
+    (df_eqs["longitude"] >= 91)
+    & (df_eqs["longitude"] <= 134)
+    & (df_eqs["latitude"] >= -22)
+    & (df_eqs["latitude"] <= 19)
+    & (df_eqs["magType"] != "mb")
+]
+
+fig = pygmt.Figure()
+fig.basemap(region=[91, 134, -22, 19], projection="M15c", frame=True)
+fig.coast(land="gray95", shorelines="gray50")
+
+# Plot epicenters with color (hypocentral depth) or size (moment magnitude)
+pygmt.makecpt(cmap="SCM/navia", series=[0, 700], reverse=True, transparency=30)
+fig.colorbar(frame=["xaf+lHypocentral depth", "y+lkm"])
+fig.plot(
+    x=df_eqs.longitude,
+    y=df_eqs.latitude,
+    size=0.005 * 2**df_eqs.mag,
+    fill=df_eqs.depth,
+    cmap=True,
+    style="c",
+    pen="gray10",
+)
+# Add legend for size-coding
+legend = io.StringIO(
+    "\n".join(f"S 0.4 c {0.005 * 2**mag:.2f} - 1p 1 Mw {mag}" for mag in [5, 6, 7])

```

Comment:
See my comment https://github.com/GenericMappingTools/pygmt-paper-figures/pull/5/files#r2570296868
```suggestion
    "\n".join(f"S 0.4 c {0.005 * 2**mag:.2f} - 1p 1 M {mag}" for mag in [5, 6, 7])
```